veracrypt: init at 1.19

dsferruzza opened the pull request:
###### Motivation for this change

[TrueCrypt](https://en.wikipedia.org/wiki/TrueCrypt) has been discontinued for several years.
This PR adds a package for [VeraCrypt](https://en.wikipedia.org/wiki/VeraCrypt), a maintained fork of TrueCrypt.

I used `truecrypt` [package](https://github.com/NixOS/nixpkgs/blob/master/pkgs/applications/misc/truecrypt/default.nix) as a start, and tried to create a nice package; but I'm a beginner so I guess not everything is perfect.
I would be happy to have feedback and advices!

I tested this package on NixOS.
I was able to successfully do the following steps:
- run VeraCrypt
- create a container
- mount it
- store a file in it
- unmount it
- move it to another computer running another OS with an official binary version of VeraCrypt
- mount it
- recover my file

I let the `maintainers` field empty.
I don't know if I should add myself, or @viric (the `truecrypt` package maintainer), or someone else.

###### Things done

- [ ] Tested using sandboxing
  ([nix.useSandbox](http://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS,
    or option `build-use-sandbox` in [`nix.conf`](http://nixos.org/nix/manual/#sec-conf-file)
    on non-NixOS)
- Built on platform(s)
   - [x] NixOS
   - [ ] macOS
   - [ ] Linux
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nox --run "nox-review wip"`
- [x] Tested execution of all binary files (usually in `./result/bin/`)
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

---

ndowens commented on pkgs/applications/misc/veracrypt/default.nix:
```diff
@@ -0,0 +1,43 @@
+      cd Vera*/src
+    '';
+
+  buildInputs = [ pkgconfig fuse devicemapper wxGTK30 nasm makeself ];
```
pkgconfig goes into nativeBuildInputs

dsferruzza replied:
Done!

ndowens commented on pkgs/applications/misc/veracrypt/default.nix:
```diff
@@ -0,0 +1,43 @@
+    homepage = https://veracrypt.codeplex.com/;
+    license = "VeraCrypt License";
+    maintainers = null; # TO DEFINE
+    platforms = with stdenv.lib.platforms; linux;
```
Should be platforms = platforms.linux since it is only one platform

dsferruzza replied:
Done!

dsferruzza commented:
It seems the first change broke the build: https://travis-ci.org/NixOS/nixpkgs/jobs/216476198#L1575

@ndowens Any idea of why it happened?
I tried it on my machine but it didn't rebuild the package so I assumed it didn't need to because the result would be the same.
Also I couldn't find the difference between `buildInputs` and `nativeBuildInputs`.

Mic92 commented on pkgs/applications/misc/veracrypt/default.nix:
```diff
@@ -0,0 +1,44 @@
+    description = "Free Open-Source filesystem on-the-fly encryption";
+    homepage = https://veracrypt.codeplex.com/;
+    license = "VeraCrypt License";
+    maintainers = null; # TO DEFINE
```
Do you want to maintains this package?

dsferruzza replied:
What does it implies exactly?
Updating the package when there is an update upstream?

joachifm replied:
It implies that you have some vested interest in the package; I don't think there's an implication of a certain level of commitment, really, but you can expect to be pinged if there's an issue.

dsferruzza replied:
OK I understand!
So I shall add myself in `lib/maintainers.nix` and update the field in the package, right?

joachifm replied:
That'd be great :)

dsferruzza replied:
Done!

ndowens commented:
The change didn't break it. Travis seems to be broken again. The difference between the buildinputs is that native is is used for building the pkg vs buildinputs is for what the pkg uses for runtime

dsferruzza commented:
@ndowens I don't believe that the names perfectly describe the semantics, but thanks for the explanation I'll keep that in mind :)

joachifm commented on pkgs/applications/misc/veracrypt/default.nix:
```diff
@@ -0,0 +1,44 @@
+      cd Vera*/src
+    '';
+
+  nativeBuildInput = [ pkgconfig ];
```
Typo: should be `nativeBuildInputs`

dsferruzza replied:
Oh indeed! I fixed it

joachifm commented:
The travis error is legit.

joachifm commented:
Added a few things for you to consider.

dsferruzza commented:
Thanks I learned a few tips ^_^

dsferruzza committed:
maintainers: add dsferruzza

dsferruzza committed:
veracrypt: init at 1.19

joachifm committed:
veracrypt: refactorings

- Conditionally add wxGTK30 to inputs to avoid
  eval'ing it regardless of wxGUI
- Patch full path to exe into desktop file
- Move some inputs to native
- Do not write intermediate tar file on unpack

joachifm commented:
Thank you

dsferruzza commented:
Thank you for helping me ^_^

Do you now if/when the package will be available in a release channel?

joachifm commented:
It's already in the unstable channel, but hasn't been picked to the release branch.

dsferruzza commented:
OK!
So how does it work? Do I need to wait until the next release freeze or is there another way?

joachifm commented:
You can submit a PR against the release branch, containing cherry-picked patches (please use `-x` to retain a reference to the original commit).
